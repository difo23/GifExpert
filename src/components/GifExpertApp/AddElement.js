import React, { useState } from 'react';
import PropTypes from 'prop-types';
const AddElement = ({ handleAdd }) => {
	const [ element, setElement ] = useState('');

	const handleChange = (e) => {
		setElement(e.target.value);
	};

	const handleSumit = (e) => {
		e.preventDefault();

		if (element.trim().length > 2) {
			handleAdd((category) => [ ...category, element ]);
		}

		setElement('');
	};

	return (
		<div>
			<p>{element}</p>
			<form onSubmit={handleSumit}>
				<div>
					<input type="text" value={element} onChange={handleChange} />
				</div>
			</form>
		</div>
	);
};

AddElement.propTypes = {
	handleAdd: PropTypes.func.isRequired
};

export default AddElement;
