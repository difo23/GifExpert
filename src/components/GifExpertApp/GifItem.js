import React from 'react';
import PropTypes from 'prop-types';

const GifItem = ({ key, title, url }) => {
	return (
		<li key={key} className="card animate__animated animate__fadeIn">
			<h4>{title}</h4>
			<img src={url} alt={title} />
		</li>
	);
};

GifItem.propTypes = {
	title: PropTypes.string,
	url: PropTypes.string
};

export default GifItem;
