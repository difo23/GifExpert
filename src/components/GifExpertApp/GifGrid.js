import React /*,{ useState, useEffect }*/ from 'react';
import PropTypes from 'prop-types';
import { useFetchGifs } from '../../hooks/useFetchGifs';
import { LinearProgress } from '@material-ui/core';
import GifItem from './GifItem';

const GifGrid = ({ category }) => {
	const { data, loading } = useFetchGifs(category);

	return (
		<div>
			<h3>{category}</h3>
			{loading && (
				<p id="loading">
					<LinearProgress />
				</p>
			)}
			<ul className="card-grid">
				{data.map(({ id, url, title }) => {
					return <GifItem key={id} url={url} title={title} />;
				})}
			</ul>
		</div>
	);
};

GifGrid.propTypes = {
	category: PropTypes.string
};

export default GifGrid;
