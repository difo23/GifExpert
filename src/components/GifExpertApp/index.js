import React, { /*PropTypes,*/ useState } from 'react';
import AddElement from './AddElement';
import GifGrid from './GifGrid';

const GifExpertApp = ({ defaultCategories = [] }) => {
	const [ category, setCategory ] = useState(defaultCategories);

	return (
		<div>
			<h2> GifExpertApp</h2>
			<hr />
			<AddElement handleAdd={setCategory} />
			<div>{category.map((element) => <GifGrid key={element} category={element} />)}</div>
		</div>
	);
};

//GifExpertApp.propTypes = {};

export default GifExpertApp;
