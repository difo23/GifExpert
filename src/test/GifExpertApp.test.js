import React from 'react';
import { shallow } from 'enzyme';
import GifExpertApp from '../components/GifExpertApp';

describe('Pruebas componente App index', () => {
	const defaultCategories = [ 'Hola', 'Mono' ];
	const wrapper = shallow(<GifExpertApp defaultCategories={defaultCategories} />);

	test('should snapshot', () => {
		expect(wrapper).toMatchSnapshot();
	});

	test('should ', () => {
		expect(wrapper.find('GifGrid').length).toBe(defaultCategories.length);
	});
});
