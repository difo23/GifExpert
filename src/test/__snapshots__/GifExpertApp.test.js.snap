// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas componente App index should snapshot 1`] = `
<div>
  <h2>
     GifExpertApp
  </h2>
  <hr />
  <AddElement
    handleAdd={[Function]}
  />
  <div>
    <GifGrid
      category="Hola"
      key="Hola"
    />
    <GifGrid
      category="Mono"
      key="Mono"
    />
  </div>
</div>
`;
